extensions/resource-deployment/src/ui/notebookWizard/notebookWizardPage.ts

```diff
@@ -96,7 +96,12 @@ export class NotebookWizardPage extends ResourceTypePage {
 			this.wizard.wizardObject.generateScriptButton.hidden = true;
 		}
 
+		// by default the generate button is disabled and will wait for the page validation to be enabled.
+		this.wizard.wizardObject.generateScriptButton.enabled = false;
```

Comment:
Did you verify that this works correctly when you first enter a page that had empty required fields? My observation was that we were running the validations as soon as the pages were created (when the wizard was first opened) and then only again if anything on the page changed - which if true would mean this would only be set to true once someone made changes on a non-summary page.